Simplified/Book/UI/NYPLBookCellDelegate+AudiobookLastListenPosition.swift

```diff
@@ -0,0 +1,123 @@
+        isActive = UIApplication.shared.applicationState == .active
+      }
+
+      // Save audiobook progress to disk if app is in background
+      if !isActive {
+        self?.savePosition()
+      }
+      
+      // Post audiobook progress to server
+      if let manager = weakManager {
+        manager.lastListenPositionSynchronizer?.syncLastListenPositionToServer()
+      }
+    }
+    manager.setLastListenPositionSyncingTimer(timer)
+  }
+  
+  @objc(setLastListenPositionSynchronizerForBook:audiobookManager:bookRegistryProvider:)
+  func setLastListenPositionSynchronizer(for book: NYPLBook,
+                                         audiobookManager: DefaultAudiobookManager,
+                                         bookRegistryProvider: NYPLBookRegistryProvider) {
+    let lastListenPosSynchronizer = NYPLLastListenPositionSynchronizer(book: book,
+                                                                       bookRegistryProvider: bookRegistryProvider,
+                                                                       annotationsSynchronizer: NYPLRootTabBarController.shared().annotationsSynchronizer)
+    
+    audiobookManager.lastListenPositionSynchronizer = lastListenPosSynchronizer
+  }
+  
+  @objc(restoreLastListenPositionAndPresentAudiobookPlayerVC:audiobookManager:successCompletion:)
+  func restoreLastListenPositionAndPresent(audiobookPlayerVC: AudiobookPlayerViewController,
+                                           audiobookManager: DefaultAudiobookManager,
+                                           successCompletion: @escaping () -> ()) {
+    // Restore last listen position from local storage and server
+    audiobookManager.lastListenPositionSynchronizer?.getLastListenPosition(completion: { [weak self] localPosition, serverPosition in
+      
+      guard let self = self else {
+        return
+      }
+      
+      NYPLMainThreadRun.asyncIfNeeded {
+        // Present audio player
+        NYPLRootTabBarController.shared().pushViewController(audiobookPlayerVC, animated: true)
+        // Call completion handler when audiobook has been successfully opened
+        successCompletion()
+        
+        // Present alert for user to decide if they want to move to position found on server
+        if let serverPosition = serverPosition {
+          self.presentAlert(serverPosition) { position in
+            let finalPosition = position != nil ? position : localPosition
+            guard let finalPosition = finalPosition else {
+              return
+            }
+            
+            audiobookManager.movePlayhead(to: finalPosition)
+          }
+        } else if let localPosition = localPosition {
+          audiobookManager.movePlayhead(to: localPosition)
+        }
+      }
+    })
+  }
+  
+  private func presentAlert(_ serverPosition: NYPLAudiobookBookmark,
+                            completion: @escaping (NYPLAudiobookBookmark?) -> ()) {
+    // TODO: - Update localized strings
```

Comment:
@ettore I am checking with Risa about the wording, will update this and also reach out to you about the Italian version of it.